Course/Course/Presentation/Container/CourseContainerViewModel.swift

```diff
@@ -140,6 +143,9 @@ public class CourseContainerViewModel: BaseCourseViewModel {
         do {
             if isInternetAvaliable {
                 courseStructure = try await interactor.getCourseBlocks(courseID: courseID)
+                Task {
```

Comment:
On the course home screen, we are executing the `getCourseDates` API call solely to determine whether to display an outdated calendar alert or not. There is no UI population involving dates at this stage. Hence, we can optimize the loading time of this screen by running the `getCourseDates` API call on a separate thread. Once we receive the response regarding the dates, we can then decide whether to present the outdated alert or not.

Additionally, it's essential to populate `getCourseBlocks` before invoking `getCourseDates` because the `getCourseDates` method involves offline population of `courseStructure`.